boa/src/syntax/parser/error.rs

```diff
@@ -98,6 +98,7 @@ impl ParseError {
     }
 
     /// Creates a new `Unimplemented` parsing error.
+    #[allow(dead_code)]
```

Comment:
Hmmm interesting, don't we have any unimplemented parsing constructs?

Reply:
Certainly, but the only call to this function was in `ForInLoop`. Maybe it could be generalized?

Reply:
Yes, I think it's good like this for now, at some point we should review all the unimplemented syntax and add calls to this in those places.